sqlglot/dialects/databricks.py

```diff
@@ -23,6 +25,21 @@ class Parser(Spark.Parser):
             TokenType.COLON: exp.JSONExtract,
         }
 
+        def _parse_generated(self, kind: t.Optional[str] = None) -> exp.ComputedColumnConstraint:
```

Comment:
should this be in hive or spark class instead or is this databricks only?

Reply:
Why define a new method for this? We could also reuse `_parse_generated_as_identity` and then construct the appropriate `ComputedColumnConstraint` instance from it to avoid duplication.

Reply:
It should be Spark and Databricks (not Spark2), see https://github.com/apache/spark/pull/38823